src/data/fin/vec_notation.lean

```diff
@@ -274,14 +274,12 @@ end val
 
 section smul
 
-variables [semiring α]
--- TODO: if I generalize these lemmas to `[has_scalar M α]`, then Lean fails to apply them
--- in `data.complex.module`
```

Comment:
This is still true, but IMO isn't a good enough reason to justify not generalizing.

Reply:
Do we have an application where we need this generalisation?

I'd rather have the definite benefit of a `simp` lemma firing than the possible benefit of a generalisation being used some day.

Reply:
I was trying to use it in the cross product PR, but it didn't match in its current form.

This is the same issue as https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/has_add.20versus.20add_semigroup/near/267127641, which seems to be a deeper lean problem.

If we really care about having the simp lemma, we could add a new `smul_cons_semiring` lemma that has the special case for simp, but that might interfere with debugging the underlying problem in future.

Reply:
Given that we in fact do have an application (the cross product PR) I think we should just go ahead with this.